### PR TITLE
[Intake] Gating behind feature flag and permissions

### PR DESCRIPTION
**Pull request by NickHeiner**
Connect #3434.

##  To test
1. See that an Intake user shows up on the Switch Users screen.
1. See that Intake shows up as an app on the Switch Users screen.
1. See that you can't access the `/intake/*` URLs unless you're the Intake user

~~I am not sure that I did the feature flag gating correctly.~~

### CODE REVIEW

File: app/controllers/intake_controller.rb [hunk 1]
```diff
@@ -1,4 +1,18 @@
 class IntakeController < ApplicationController
+  before_action :verify_access, :react_routed, :verify_feature_enabled
+
+  def set_application
+    RequestStore.store[:application] = "intake"
```
Comment (NickHeiner) [inline app/controllers/intake_controller.rb]: This is not used yet, but it seems like a good thing to do.

File: app/controllers/intake_controller.rb [hunk 2]
```diff
@@ -1,4 +1,18 @@
 class IntakeController < ApplicationController
+  before_action :verify_access, :react_routed, :verify_feature_enabled, :set_application
+
+  def set_application
+    RequestStore.store[:application] = "intake"
```
Comment (NickHeiner) [inline app/controllers/intake_controller.rb]: This is not used yet, but it seems like a good thing to do.

File: app/controllers/intake_controller.rb [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+  end
+
+  def verify_access
+    verify_authorized_roles("Intake")
+  end
+
+  def verify_feature_enabled
```
Comment (NickHeiner) [inline app/controllers/intake_controller.rb]: I'm not sure I did this right. I don't know what the preferred way to test it is (manually toggle in Rails console?).

File: app/controllers/intake_controller.rb (hunk 2 above)
Comment (shanear) [reply app/controllers/intake_controller.rb]: This is so we show the right icon i believe?

File: app/controllers/intake_controller.rb (hunk 3 above)
Comment (shanear) [reply app/controllers/intake_controller.rb]: Yep, this looks good. Yeah we manually toggle it in the Rails console (which updates Redis)

File: app/models/user.rb [hunk 4]
```diff
@@ -9,7 +9,7 @@ class User < ActiveRecord::Base
   attr_writer :regional_office
 
   FUNCTIONS = ["Establish Claim", "Manage Claim Establishment", "Certify Appeal",
-               "Reader", "Hearing Prep"].freeze
+               "Reader", "Hearing Prep", "Intake"].freeze
```
Comment (shanear) [inline app/models/user.rb]: this is "Mail Intake" i think

File: app/controllers/intake_controller.rb [hunk 5]
```diff
@@ -1,4 +1,18 @@
 class IntakeController < ApplicationController
+  before_action :verify_access, :react_routed, :verify_feature_enabled, :set_application
+
+  def set_application
+    RequestStore.store[:application] = "intake"
+  end
+
+  def verify_access
+    verify_authorized_roles("Intake")
```
Comment (shanear) [inline app/controllers/intake_controller.rb]: should be "Mail Intake"

**Pull request merged**